### DISCUSSION TITLE
mysql plugin

### DISCUSSION THREAD

**Discussion by dmitriy-udoratin** [tg123/sshpiper#187] [Q&A]
Hey, mentions of mysql plugin in the documentation and code are only in v0 branch, 

- does this mean that the latest versions 1.X do not support it?
- is any migration available when switching from yaml driver to mysql plugin?

**Comment by tg123** [comment]
did not get a chance to migrate db plugin

is k8s plugin good for you? you can manage pipes using kubectl

**Comment by dmitriy-udoratin** [comment]
I've tested k8s and yes, it seems to be working fine, in a similar way as yaml.
I just thought if there are any benefits using mysql

**Comment by dmitriy-udoratin** [comment]
regarding k8s, questions that come to mind so far is
- what will happen to performance if there are thousands of pipes
- how it is convenient to back up

**Reply by tg123** [reply]
wow thousands? working k8s with ~1.2k 

you can dump this way
```
kubectl get pipes -o yaml > backup.yaml
kubectl apply -f backup.yaml
```

let me see if i have time to port db plugin

**Reply by dmitriy-udoratin** [reply]
> wow thousands?

there are 400-500 ssh users at the moment so it would be 400-500 pipes in k8s plugin terms. but in the long term it might be up to a thousand or more users

there is a task of migration from yaml driver, so now I'm thinking if it makes sense to migrate to the k8s plugin or is it better to serve this by mysql plugin and so wait until it's ready in v1

**Reply by tg123** [reply]
internally working with up to 4k pipes with 1 32G api server + 3 ssd etcd
prefer k8s now

that is why did not port db plugin to v1

**Reply by dmitriy-udoratin** [reply]
ok thanks, will try k8s
